ngraph/core/reference/include/ngraph/runtime/reference/convolution.hpp

```diff
@@ -171,6 +171,81 @@ namespace ngraph
                             std::prev(filter_shape.end(), spatial_rank), missing_dims, 1);
                     }
                 }
+
```

Comment:
Added validation functions to check input ranks, spatial dimensions of strides, dilations and pads. Additionally output shape is inferred and compared to output shape provided by the tensor descriptor. These checks are introduced because when using dynamic shapes some node validation check from `validate_and_infer_types` are by-passed.